pyomo/core/base/units_container.py

```diff
@@ -1409,6 +1409,17 @@ def convert(self, src, to_units=None):
         src_pyomo_unit, src_pint_unit = self._get_units_tuple(src)
         to_pyomo_unit, to_pint_unit = self._get_units_tuple(to_units)
 
+        # check if they are both dimensionless
+        src_dimensionless = \
+            _UnitExtractionVisitor(self)._pint_unit_equivalent_to_dimensionless(src_pint_unit)
+        to_dimensionless = \
+            _UnitExtractionVisitor(self)._pint_unit_equivalent_to_dimensionless(to_pint_unit)
+        if src_dimensionless and to_dimensionless:
+            return src
+        elif src_dimensionless or to_dimensionless:
+            raise InconsistentUnitsError(src_pint_unit, to_pint_unit,
+                                         'Error in convert: units not compatible.')
```

Comment:
Consistency question: below the exception is:
```python
raise UnitsError('Cannot convert {0:s} to {1:s}. Units are not compatible.'.format(str(src_pyomo_unit), str(to_pyomo_unit)))
```
Is there a reason that this raises a different message?